dirty_cat/datasets/fetching.py

```diff
@@ -166,6 +166,71 @@ def fetch_openml_dataset(
     }
 
 
+def fetch_world_bank_data(dataset_id: str, indicator: str,
+                          data_directory: Path = get_data_dir(),
+                          ) -> Dict[str, Any]:
+    """
+    Gets a dataset from World Bank open data platform
+    (https://data.worldbank.org/).
+
+    Parameters
+    ----------
+    dataset_id: str
+        The ID of the dataset to fetch.
+    indicator: str
+        Name of the indicator to extract.
+    data_directory: Path
+        Optional. A directory to save the data to.
+        By default, the dirty_cat data directory.
+
+    Returns
+    -------
+    Dict[str, Any]
+        A dictionary containing:
+          - ``description``: str
+              The description of the dataset,
+              as gathered from World Bank data.
+          - ``source``: str
+              The dataset's URL from the World Bank data platform.
+          - ``path``: pathlib.Path
+              The local path leading to the dataset,
+              saved as a CSV file.
+
+    """
+
+    # Make path absolute
+    if not os.path.exists(data_directory):
+        os.makedirs(data_directory)
+    data_directory = data_directory.resolve() / 'world_bank.zip'
+    zip_path = Path(str(data_directory) + '_folder')
+    # Download the file :
+    url = f'https://api.worldbank.org/v2/en/indicator/{dataset_id}?downloadformat=csv'
+    urllib.request.urlretrieve(url, data_directory)
+    # Extract csv file :
+    with ZipFile(data_directory, 'r') as f:
+        names = f.namelist()
+        for n in names:
+            if 'Metadata' not in n:
+                f.extract(
+                    n, path=zip_path)
```

Comment:
Did black format it like that ? I think it should fit on a single line (sorry for pointing out syntax, this is counter-productive)
```suggestion
                f.extract(n, path=zip_path)
```